scripts/release/build/build_algod_docker.sh

```diff
@@ -38,9 +38,9 @@ DOCKER_IMAGE="algorand/algod_${CHANNEL_VERSION}:${DOCKER_TAG}"
 RESULT_DIR="${HOME}/node_pkg/"
 DOCKERFILE="${HOME}/go/src/github.com/algorand/go-algorand/docker/build/algod.Dockerfile"
 START_ALGOD_FILE="start_algod_docker.sh"
-# Use go version specified by get_golang_version.sh
+# Use go build version specified by get_golang_version.sh
 pushd "${HOME}/go/src/github.com/algorand/go-algorand"
-if ! GOLANG_VERSION=$("./scripts/get_golang_version.sh")
+if ! GOLANG_VERSION=$("./scripts/check_golang_version.sh")
```

Comment:
Semantically, I'm a little bit bothered by this. We're defining the version of golang we want with a script named 'check_golang_version.sh', not 'get_golang_version.sh'. I think splitting the check and get functionalities are fine, but it makes uses such as this cognitively difficult.

Reply:
I agree, I've always been bothered by this haha
I'll look into making it less cognitively jarring.

Reply:
I just pushed up a commit that changes the checks in all the scripts that use it.  The new idiom looks like this:
```
if ! ./scripts/check_golang_version.sh
then
    exit 1
fi
# Get the go build version.
GOLANG_VERSION=$(./scripts/get_golang_version.sh)
```
It's more expensive, but I think the benefit is that it reads a lot better.  The previous way had bugged me because the old version of `get_golang_version.sh` was pulling double-duty in a not-so-obvious way from a cursory read of the code.

For example, it wasn't just "getting" the golang version, it was also checking it.  And if the check failed the calling script would echo the return value captured in the `GOLANG_VERSION` variable before exiting.  That was just weird.

Reply:
I agree - longer code in this case makes more sense, and it's not _that_ much more expensive, as it's a build env execution, not a run time execution.